File: src/old-java/com/rwbymod/item/ItemDust.java
package com.rwbymod.item;

import com.rwbymod.misc.EnumDustStimulus;

import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;

public class ItemDust extends ItemDustContainer {
	
	public ItemDust() {
		super("dust");
		this.setExplosionStimuli(EnumDustStimulus.item);
		this.setExplosionStrength(1f);
		this.setExplosionChance(50);
	}

	@Override
	protected void onExplosion(ItemStack item, Entity entity, EnumDustStimulus stimulus) {
		if(entity instanceof EntityPlayer) {
			if(!((EntityPlayer)entity).isCreative()) item.setCount(0); 
		}
	}

}

File: src/main/scala/com/rwbymod/item/ItemDust.scala
package com.rwbymod.item

import net.minecraft.entity.player.EntityPlayer
import com.rwbymod.dust.EnumDustStimulus

class ItemDust extends ItemDustContainer("dust", explosionStimuli = EnumDustStimulus.item) {

  override def onExplosion(context:ExplosionContext) =
    if(context._2.isInstanceOf[EntityPlayer] && !context._2.asInstanceOf[EntityPlayer].isCreative) context._1.setCount(0)

}